app/webhook.go

```diff
@@ -323,6 +340,21 @@ func (a *App) CreateIncomingWebhookForChannel(creatorId string, channel *model.C
 		return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_username.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	if len(hook.WhiteIpList) > 0 {
+		if !(model.IsValidIpList(hook.WhiteIpList)) {
+			return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_white_ip_list.app_error", nil, "", http.StatusBadRequest)
+		}
+	} else if len(hook.HmacAlgorithm) > 0 {
```

Comment:
Are the whitelist and signatures intentionally mutually exclusive?

Reply:
Yes. If i'm not wrong, for whitelist, senders usually share the ips of their load balancers and cluster ips which send out webhooks. In this case, signature may not be needed. In my understanding, signature based authentication used when whitelist is not an option.

Reply:
Yes. If i'm not wrong, for whitelist, senders usually share the ips of their load balancers and cluster ips which send out webhooks. In this case, signature may not be needed. In my understanding, signature based authentication used when whitelist is not an option.